### ISSUE TITLE
Enable not validating DTD

### ISSUE THREAD

**Issue by blacelle** [revelc/xml-formatter#9]
In some cases, the internet is not available (e.g. corporate proxy). We then encounter the following kind of errors:

```
Caused by: java.net.UnknownHostException: www.puppycrawl.com
    at java.net.AbstractPlainSocketImpl.connect (AbstractPlainSocketImpl.java:184)
    at java.net.PlainSocketImpl.connect (PlainSocketImpl.java:172)
    at java.net.SocksSocketImpl.connect (SocksSocketImpl.java:392)
    at java.net.Socket.connect (Socket.java:589)
    at java.net.Socket.connect (Socket.java:538)
    at sun.net.NetworkClient.doConnect (NetworkClient.java:180)
    at sun.net.www.http.HttpClient.openServer (HttpClient.java:463)
    at sun.net.www.http.HttpClient.openServer (HttpClient.java:558)
    at sun.net.www.http.HttpClient.<init> (HttpClient.java:242)
    at sun.net.www.http.HttpClient.New (HttpClient.java:339)
    at sun.net.www.http.HttpClient.New (HttpClient.java:357)
    at sun.net.www.protocol.http.HttpURLConnection.getNewHttpClient (HttpURLConnection.java:1220)
    at sun.net.www.protocol.http.HttpURLConnection.plainConnect0 (HttpURLConnection.java:1156)
    at sun.net.www.protocol.http.HttpURLConnection.plainConnect (HttpURLConnection.java:1050)
    at sun.net.www.protocol.http.HttpURLConnection.connect (HttpURLConnection.java:984)
    at sun.net.www.protocol.http.HttpURLConnection.getInputStream0 (HttpURLConnection.java:1564)
    at sun.net.www.protocol.http.HttpURLConnection.getInputStream (HttpURLConnection.java:1492)
    at com.sun.org.apache.xerces.internal.impl.XMLEntityManager.setupCurrentEntity (XMLEntityManager.java:647)
    at com.sun.org.apache.xerces.internal.impl.XMLEntityManager.startEntity (XMLEntityManager.java:1304)
    at com.sun.org.apache.xerces.internal.impl.XMLEntityManager.startDTDEntity (XMLEntityManager.java:1270)
    at com.sun.org.apache.xerces.internal.impl.XMLDTDScannerImpl.setInputSource (XMLDTDScannerImpl.java:264)
    at com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl$DTDDriver.dispatch (XMLDocumentScannerImpl.java:1161)
    at com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl$DTDDriver.next (XMLDocumentScannerImpl.java:1045)
    at com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl$PrologDriver.next (XMLDocumentScannerImpl.java:959)
    at com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl.next (XMLDocumentScannerImpl.java:602)
    at com.sun.org.apache.xerces.internal.impl.XMLNSDocumentScannerImpl.next (XMLNSDocumentScannerImpl.java:112)
    at com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl.scanDocument (XMLDocumentFragmentScannerImpl.java:505)
    at com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse (XML11Configuration.java:842)
    at com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse (XML11Configuration.java:771)
    at com.sun.org.apache.xerces.internal.parsers.XMLParser.parse (XMLParser.java:141)
    at com.sun.org.apache.xerces.internal.parsers.AbstractSAXParser.parse (AbstractSAXParser.java:1213)
    at com.sun.org.apache.xerces.internal.jaxp.SAXParserImpl$JAXPSAXParser.parse (SAXParserImpl.java:643)
    at net.revelc.code.formatter.xml.lib.XmlDocumentFormatter.validateWellFormedness (XmlDocumentFormatter.java:115)
    at net.revelc.code.formatter.xml.lib.XmlDocumentFormatter.format (XmlDocumentFormatter.java:77)
    at net.revelc.code.formatter.xml.XMLFormatter.doFormat (XMLFormatter.java:54)
    at net.revelc.code.formatter.AbstractCacheableFormatter.formatFile (AbstractCacheableFormatter.java:47)
    at net.revelc.code.formatter.FormatterMojo.doFormatFile (FormatterMojo.java:530)
    at net.revelc.code.formatter.FormatterMojo.formatFile (FormatterMojo.java:464)
    at net.revelc.code.formatter.FormatterMojo.execute (FormatterMojo.java:344)
```

Would it be possible to add an option to skip XML validation, as suggested in https://stackoverflow.com/questions/4002885/unknown-host-exception-while-parsing-an-xml-file ?

**Comment by blacelle** [comment]
It seems related with https://github.com/revelc/xml-formatter/pull/8/files

**Comment by jam01** [comment]
Hey @blacelle can you include the entire mvn output? I'm not entirely sure that the formatter does any validation based on DTDs or XSDs

**Comment by hazendaz** [comment]
I believe it is on XML formatting due to newer updates.  I thought a release was pushed to  allow for disable though.  Are you using the latest release?

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by blacelle** [comment]
@jam01 The issue happens in the context of a corporate customer. I would not take the risk of copy-pasting too much information. This way happening in the format goal, not in the check one.

@hazendaz I spot no option regarding validation in https://github.com/revelc/xml-formatter/blob/master/src/main/java/net/revelc/code/formatter/xml/lib/FormattingPreferences.java then I throught there was no alraedy existing option. We use https://github.com/revelc/formatter-maven-plugin/releases/tag/formatter-maven-plugin-2.10.0 which is the latest version of the maven plugin. However, I did not checked it relies on the latest version of revelc/xml-formatter

**Comment by ctubbsii** [comment]
If this is the same issue as in revelc/xml-formatter#8 , then we're just waiting on @jam01 to merge that, so we can release the updated version which uses it. I could merge it, but I'm trying not to be a single point of failure. :smiley_cat:

**Comment by jam01** [comment]
Thanks @ctubbsii I've been meaning to catch up on this! 

I had no idea that SaxParser validated DTDs, I haven't had any XMLs that referenced them before. Anyways, I'll try to find some time to go over that PR and get it merged tomorrow.

Might make sense to add something like this https://stackoverflow.com/a/155874/4814697 to retain the valid XML checks and entirely disregard DTDs which feels like a non-goal for a formatter.

**Comment by jam01** [comment]
Fixed with https://github.com/revelc/xml-formatter/commit/1f40161714fbb94dcbe8d14412dbb881a996dd12
